gptme/message.py

```diff
@@ -126,39 +124,24 @@ def _content_files_list(
                 )
                 continue
 
-            if provider == "anthropic":
-                content.append(
-                    {
-                        "type": "image",
-                        "source": {
-                            "type": "base64",
-                            "media_type": media_type,
-                            "data": data,
-                        },
-                    }
-                )
-            elif provider == "openai":
-                # OpenAI format
-                content.append(
-                    {
-                        "type": "image_url",
-                        "image_url": {"url": f"data:{media_type};base64,{data}"},
-                    }
-                )
+            file_content = get_model().prepare_file(media_type, data)
+            if file_content:
+                content.append(file_content)
```

Comment:
I would remove this from here entirely, and move it into each provider.

Reply:
I did something. Let me know if it's what you had in mind.